### PR TITLE
feat: migrate a eslint config to support a new version

### PR DESCRIPTION
**Pull request by kilatib**
## Description 
This PR corrects this kind of notice for eslint 

More details:
https://eslint.org/docs/latest/use/configure/migration-guide#configuring-language-options
```
Oops! Something went wrong! :(

ESLint: 9.15.0

A config object is using the "env" key, which is not supported in flat config system.

Flat config uses "languageOptions.globals" to define global variables for your files.

Please see the following page for information on how to convert your config object into the correct format:
https://eslint.org/docs/latest/use/configure/migration-guide#configuring-language-options

If you're not using "env" directly (it may be coming from a plugin), please see the following:
https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config
```

## How to reproduce 
Run eslint with current config 
```
eslint -c ./index.js 
```

### CODE REVIEW

**Review by yairkukielka — CHANGES_REQUESTED** [review]
This is a breaking change: You are "migrating" the eslint config file from the legacy config ( eslint <= 8.57 ) file to the [new flat config ](https://eslint.org/blog/2024/04/eslint-v9.0.0-released/)(eslint >= 9).  If someone using eslint 8 (most of our apps) installs the next version that would be release after this PR, it won't work.

But more importantly, if you are getting this error message 
```
Oops! Something went wrong! :(

ESLint: 9.15.0

A config object is using the "env" key, which is not supported in flat config system.

Flat config uses "languageOptions.globals" to define global variables for your files.
```
It's very likely that you are using ESLint 9 in a consumer node application. This package (`@oat-sa/eslint-config-tao`) is not ready for node applications with eslint 9, as I mentioned [here](https://oatsa.slack.com/archives/CHV6WAE90/p1728040983620209). So for now, you could use eslint 8 (or, better, 8.57) in your app to unblock you and fix your current problem.

That said, we should release a new version  of `@oat-sa/eslint-config-tao` (probably major version) to migrate it to flat config (eslint >= 9), but that change should be carefully planned. We already did this for frontend apps, we just need to do it for backend apps [here](https://github.com/oat-sa/eslint-config-tao/pull/37). There are many little details to take into account for this. I would create a Jira ticket to do this properly and lay out what needs to be done there. Also, testing this kind of PRs takes time from the reviewers to make sure old apps still work with the new changes, and we need a ticket to log that time.

As a reference PR, I would recommend to take a loot at [this](https://github.com/oat-sa/eslint-config-tao/pull/37) one. The readme and changelog files should be updated clearly stating what has changed.

My recommendation would be to keep consistency with what was done in the release 4.0 of this package with the frontend (svelte) eslint migration to eslint 9:

- rename an `old.js` config to `old-legacy.js`. Example: `node.js` should be migrated to `node-legacy.js`, and `cypress.js` should be migrated to `cypress-legacy.js`. This makes clear that if anyone still wants to use the old configs, they should be thinking in migrating soonish to the new ones.
- new config files should be added to the `flat` folder. Example: `flat/node.mjs` or `flat/cypress.mjs`. This way it's clear which ones use flat config files and we still have the old config files for apps that still use eslint 8. Maybe in a next release we could remove the old legacy config files, but for now I would keep both config files, to ease migration for consumers.
- I would also design this as composable flat configs, as I did for the [svelte config]([url](https://github.com/oat-sa/eslint-config-tao/blob/master/flat/svelte.mjs)). As you can see there, we export several configs so the consumer application can choose which parts they want to use (Example 1: svelte with jest rules, Example 2: svelte with vitest rules).
- I'm not sure about what we want to do with the default config file of this package (`index.js`). It might be ok to just update it to flat config, but this would clearly be a breaking change. 

And a couple of details about your PR:
-

File: index.js [hunk 1]
```diff
@@ -1,44 +1,40 @@
...
+            mocha: true,
+            Cypress: true,
+            cy: true,
+        },
     },
-    globals: {
-        Cypress: true,
-        cy: true
-    },
-    parser: '@babel/eslint-parser',
```
Comment (yairkukielka) [inline index.js:12]: in case we are migrat

Comment (yairkukielka) [inline index.js:12]: removing the parser would be a breaking change for many apps that use babel for linting

File: index.js [hunk 2]
```diff
@@ -1,44 +1,40 @@
...
-    parserOptions: {
-        sourceType: 'module',
-        requireConfigFile: false
-    },
-    extends: 'eslint:recommended',
```
Comment (yairkukielka) [inline index.js:17]: I would keep the eslint recommended rules, as we do [here](https://github.com/oat-sa/eslint-config-tao/blob/master/flat/svelte.mjs#L15) for the frontend (svelte) config. In this PR you are removing those rules

**Comment by kilatib** [comment]
migrate to old eslint for now decline that PR

**Pull request closed without merging**